src/ShiftedProximalOperators.jl

```diff
@@ -46,12 +46,33 @@ end
 
 Set the trust-region radius of a shifted proximable function to Δ.
 This method updates the indicator of the trust region that is part of ψ.
+
+    set_bounds!(ψ, l, u)
+
+Set the trust-region l and u parameters if it is defined as a box [l,u].
 """
 function set_radius!(ψ::ShiftedProximableFunction, Δ::R) where {R <: Real}
   ψ.Δ = Δ
   return ψ
 end
 
+function set_bounds!(ψ::ShiftedProximableFunction, l, u)
+  if isa(l, Real) && isa(u, Real)
```

Comment:
Isn't there an error here if I first defined `ψ` with `l` and `u` as vectors and I call `set_bounds!(ψ, -2, 2)`?

The function should probably be something like (not tested):
```julia
function set_bounds!(ψ::ShiftedWithBox{R, V0, V1, V2, V3, V4}, l::V3, u::V4) where {R, V0, V1, V2, V3, V4}
  ψ.l .= l
  ψ.u .= u
  ψ
end
```
where
```julia
abstract ShiftedWithBox <: ShiftedProximableFunction
mutable struct ShiftedNormL0Box{ ... } <: ShiftedWithBox
mutable struct ShiftedNormL1Box{ ... } <: ShiftedWithBox
```

Reply:
I believe there is no error when `l` and `u` are defined as vectors, it is just overwritten and their type are modified. I wrote the function in that way in order to avoid errors raised by `ψ.l .= l` when `ψ.l` i initially defined as a real and `l` is a real. To be precise the error is the following : `MethodError: no method matching copyto!(::Float64, ::Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{0}, Tuple{}, typeof(identity), Tuple{Float64}})`.

Reply:
Do you prefer this syntax ? It raises no error : `isa(l,Real) ? ψ.l=l : (isa(ψ.l,Real) ? ψ.l=l : ψ.l.=l)`